ElectronicObserverTypes/Attacks/Specials/NagatoSpecialAttack.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using ElectronicObserverTypes.Extensions;
+
+namespace ElectronicObserverTypes.Attacks.Specials;
+
+public record NagatoSpecialAttack : SpecialAttack
+{
+	public NagatoSpecialAttack(IFleetData fleet) : base(fleet)
+	{
+	}
+
+	public override string GetDisplay() => Fleet.MembersInstance.ToList().FirstOrDefault() switch
+	{
+		{ MasterShip.ShipId: ShipId.NagatoKaiNi } => AttackResources.SpecialNagato,
+		{ MasterShip.ShipId: ShipId.MutsuKaiNi } => AttackResources.SpecialMutsu,
+		_ => "???"
+	};
+
+	protected override double GetTriggerRate()
+	{
+		List<IShipData?> ships = Fleet.MembersInstance.ToList();
+
+		if (!ships.Any()) return -1;
+
+		IShipData? flagship = ships.First();
+		if (flagship is null) return -1;
+		if (flagship.MasterShip.ShipId is not ShipId.NagatoKaiNi and not ShipId.MutsuKaiNi) return -1;
+
+		if (flagship.HPRate <= 0.5) return -1;
+
+		int availableShipCount = ships.Count(ship => ship?.HPCurrent > 0) - Fleet.EscapedShipList.Count;
+		if (availableShipCount < 6) return -1;
+
+		IShipData? helper = ships[1];
+		if (helper is null) return -1;
+		if (helper.MasterShip.ShipType is not ShipTypes.Battleship and not ShipTypes.Battlecruiser and not ShipTypes.AviationBattleship) return -1;
+		if (helper.HPRate <= 0.25) return -1;
+
+		// todo : find formula
+		return 0;
+	}
+
+	protected override IEnumerable<SpecialAttackHit> GetAttacks()
```

Comment:
This should probably just be
```cs
protected override List<SpecialAttackHit> GetAttacks() => new()
{
...
}
```
I used to make it IEnumerable in older code but realized it doesn't really add any value.

Reply:
Changed to List